src/briefcase/debugger.py

```diff
@@ -0,0 +1,273 @@
+import dataclasses
+import enum
+import textwrap
+from datetime import datetime
+from pathlib import Path
+from typing import TextIO
+
+from briefcase.config import AppConfig
+
+
+class Debugger(enum.StrEnum):
+    PDB = "pdb"
+    DEBUGPY = "debugpy"
+
+
+class DebuggerMode(enum.StrEnum):
+    SERVER = "server"
+    CLIENT = "client"
+
+
+_DEBGGER_REQUIREMENTS_MAPPING = {
+    Debugger.PDB: [],
+    Debugger.DEBUGPY: ["debugpy~=1.8.12"],
+}
+
+
+@dataclasses.dataclass
+class DebuggerConfig:
+    debugger: Debugger
+    debugger_mode: DebuggerMode
+    ip: str
+    port: int
+
+    @property
+    def additional_requirements(self) -> list[str]:
+        return _DEBGGER_REQUIREMENTS_MAPPING[self.debugger]
+
+    @staticmethod
+    def from_app(app: AppConfig) -> "DebuggerConfig":
+        debugger = app.debugger or Debugger.PDB
+        debugger_mode = app.debugger_mode or DebuggerMode.SERVER
+        debugger_ip = app.debugger_ip or "localhost"
+        debugger_port = app.debugger_port or 5678
+
+        try:
+            debugger = Debugger(debugger)
+        except Exception:
+            raise ValueError("debugger has a wrong value")
+
+        try:
+            debugger_port = int(debugger_port)
+        except Exception:
+            raise ValueError("debugger_port has to be an integer")
+
+        return DebuggerConfig(
+            debugger,
+            debugger_mode,
+            debugger_ip,
+            debugger_port,
+        )
+
+
+def write_debugger_startup_file(
+    app_path: Path,
+    pth_folder_path: Path | None,
+    app: AppConfig,
+    path_mappings: str,
+):
+    debugger_cfg = DebuggerConfig.from_app(app)
+
+    startup_modul = "__briefcase_startup__"
+    startup_code_path = app_path / f"{startup_modul}.py"
```

Comment:
Two notes here: 

1. There's a typo on `startup_module`
2. I'd be inclined to simplify this to just `__briefcase__`. 
3. Although you're using it for debug content here, it could also be used for a `remote control` testing feature that we're potentially looking at adding. To that end, the code to install/remove the debugger bootstrap should probably be on `base.py` as a generic capability that an app can invoke under different circumastances.

Reply:
> I'd be inclined to simplify this to just `__briefcase__`.

Double-underscore names are reserved for the Python language, so `_briefcase` would be better.

Reply:
Good point - and it would also match the broad pattern of `.pth` files that are installed by venv etc as part of "environment setup."